src/Mono.Android/Android.Runtime/JavaProxyThrowable.cs

```diff
@@ -1,23 +1,64 @@
 using System;
+using System.Diagnostics;
+using System.Reflection;
+
+using StackTraceElement = Java.Lang.StackTraceElement;
 
 namespace Android.Runtime {
 
 	class JavaProxyThrowable : Java.Lang.Error {
 
 		public  readonly Exception InnerException;
 
-		public JavaProxyThrowable (Exception innerException)
-			: base (GetDetailMessage (innerException))
+		protected JavaProxyThrowable (string message, Exception innerException)
```

Comment:
This type should probably be `sealed` -- nothing subclasses it, and I can't think of anything to subclass it *ever* -- and this constructor could thus be `private`.